doc/transaction_json.rst

```diff
@@ -0,0 +1,194 @@
+..
+  Copyright (C) 2020 Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+.. _transaction_json-label:
+
+################################
+ Stored Transaction JSON Format
+################################
+
+The stored transaction format is considered unstable and may change in an
+incompatible way at any time. It will work if the same version of dnf is used
+to store and replay (or between versions as long as it stays the same).
+
+==================
+ Top-level Object
+==================
+
+``version``
+    Type: string
+
+    The version of the stored transaction format, in the form ``MAJOR.MINOR``.
+
+    ``MAJOR`` version denotes backwards incompatible changes (old dnf won't work with
+    new transaction JSON).
+
+    ``MINOR`` version denotes extending the format without breaking backwards
+    compatibility (old dnf can work with new transaction JSON).
+
+``rpms``
+    Type: an array of :ref:`rpm <rpm-label>` objects
+
+    A list of RPM packages in the transaction.
+
+``groups``
+    Type: an array of :ref:`group <group-label>` objects
+
+    A list of groups in the transaction.
+
+``environments``
+    Type: an array of :ref:`group <environment-label>` objects
+
+    A list of environment groups in the transaction.
+
+
+.. _rpm-label:
+
+==============
+ `rpm` Object
+==============
+
+``action``
+    Type: string
+
+    Possible values: ``Downgrade, Downgraded, Install, Obsoleted, Reason Change, Reinstall, Reinstalled, Removed, Upgrade, Upgraded``
+
+    The action performed on the package in the transaction.
+
+``nevra``
+    Type: string
+
+    ``NEVRA`` (``name-epoch:version-release.arch``) of the package.
+
+``reason``
+    Type: string
+
+    Possible values: ``dependency, clean, group, unknown, user, weak-dependency``
```

Comment:
Probably I overlooked something, but I cannot found description of `repo_id`.

Reply:
Sorry, I've missed that one. Added.